detectable/src/main/java/com/synopsys/integration/detectable/detectables/bitbake/collect/BuildFileFinder.java

```diff
@@ -50,9 +50,22 @@ public File findTaskDependsFile(File sourceDir, File buildDir) throws Integratio
 
     public Optional<File> findLicenseManifestFile(File buildDir, String targetImageName, BitbakeEnvironment bitbakeEnvironment) {
         try {
+            String machine = bitbakeEnvironment.getMachine().orElse(null);
+
             String machineArch = bitbakeEnvironment.getMachineArch().orElse(null);
             File licensesDir = findLicensesDir(buildDir, bitbakeEnvironment.getLicensesDirPath().orElse(null));
-            logger.debug("Checking licenses dir {} for license.manifest for {}", licensesDir.getAbsolutePath(), targetImageName);
+
+            logger.debug(
+                "Checking licenses dir {} for license.manifest for image {}, architecture {}, and machine {}",
+                licensesDir.getAbsolutePath(), targetImageName, machineArch, machine
+            );
+
+            Optional<File> licenseFile = findMostRecentLicenseManifestFileForMachine(licensesDir, machineArch, targetImageName, machine);
+            if (licenseFile.isPresent()) {
+                return licenseFile;
+            }
+            logger.debug("Did not find a machine-specific license.manifest");
+
             List<File> licensesDirContents = generateListOfFiles(licensesDir);
```

Comment:
From here on we continue with pre-existing search logic to ensure backward-compatibility with previous Yocto versions.